packages/expo-updates/android/src/main/java/expo/modules/updates/procedures/StartupProcedure.kt

```diff
@@ -245,6 +245,10 @@ class StartupProcedure(
     errorRecovery.startMonitoring(reactContext)
   }
 
+  fun onReactInstanceException(exception: Exception) {
+    errorRecovery.handleException(exception)
```

Comment:
since overriding the exception handler doesn't work on bridgeless, should we update ErrorRecovery to not register the error when in bridgeless mode?

edit: I see that the registration is done in `onDidCreateReactInstanceManager`/`onDidCreateReactInstance` so it shouldn't be running on bridgeless. let's add an assert to `registerErrorHandler` to assert non-bridgeless and also add an assert to `onReactInstanceException` (here) to assert bridgeless, just so that if upstream (expo-modules-core) is changed our assumptions hold.

Reply:
tried to split that as two different implementation, hopes that looks good

```kotlin
  private fun registerErrorHandler(reactContext: ReactContext) {
    if (ReactFeatureFlags.enableBridgelessArchitecture) {
      registerErrorHandlerImplBridgeless(reactContext)
    } else {
      registerErrorHandlerImplBridge(reactContext)
    }
  }
```